### ISSUE TITLE
import resolving issue around namespace package.

### ISSUE THREAD

**Issue by heejaechang** [microsoft/pyright#2358]
**Describe the bug**
Pyright shows an unresolvable import squiggles involving namespace package, but if F5, it runs fine.

**To Reproduce**
repo code is here
https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/tests/fourslash/completions.importSubmodule.fourslash.ts

**Expected behavior**
"setup" not showing squiggles for "from submodule.setup import ... "

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/1333179/135533096-7edbf887-d70c-4e8b-8775-f2348814b658.png)

**VS Code extension or command-line**
vs code extension

**Additional context**
it looks like due to extra path, submodule.setup is considered not resolvable even if it is considered resolvable from project root (where test.py file is). and one (extra path) that contains __ init __ wins over and show squiggles. but if run test.py, it works fine by python interpreter.

**Comment by erictraut** [comment]
I think the behavior is correct. The example is adding an extraPaths of "submodule". If you manually add that path to the PYTHONPATH environment variable before running "python test.py", you will receive an import error from the interpreter:
```
  File "/Users/eric/Desktop/Test/test.py", line 6, in <module>
    from submodule.setup import bar
```

If you don't want to edit the PYTHONPATH variable, you can also manually manipulate the sys.paths at the top of `test.py` to add the extra path at runtime.

```python
import os.path
import sys
sys.path.append(os.path.dirname(__file__) + "/submodule")
```

As before, the import `submodule.setup` cannot be resolved, consistent with the diagnostic.
